src/jit/lsra.cpp

```diff
@@ -4301,7 +4301,7 @@ void LinearScan::updateRegStateForArg(LclVarDsc* argDsc)
             isFloat = true;
         }
 
-        if (isFloat)
+        if (isFloat && !compiler->opts.compUseSoftFP)
         {
```

Comment:
@BruceForstall Yes I'm sure that with compUseSoftFP, lvIsHfaRegArg() is always false since FEATURE_HFA is disabled for ARM_SOFTFP.